src/store/note_record/input_note_record.rs

```diff
@@ -187,3 +187,22 @@ impl TryInto<InputNote> for InputNoteRecord {
         }
     }
 }
+
+impl TryInto<Note> for InputNoteRecord {
+    type Error = ClientError;
+
+    fn try_into(self) -> Result<Note, Self::Error> {
+        match self.metadata {
+            Some(metadata) => {
+                let note_inputs = NoteInputs::new(self.details.inputs)?;
+                let note_recipient =
+                    NoteRecipient::new(self.details.serial_num, self.details.script, note_inputs);
+                let note = Note::new(self.assets, metadata, note_recipient);
+                Ok(note)
+            },
+            None => Err(ClientError::NoteError(miden_objects::NoteError::invalid_origin_index(
+                "Input Note Record contains no metadata".to_string(),
+            ))),
```

Comment:
This works for now but we'll probably need to add a `NoteRecordError` to `ClientError` in the future.

Reply:
Yes, let's do this if possible. Let's add a `ClientError::NoteRecordError(String)` that explains that the note does not contain metadata and the conversion is impossible.